File: src/main/java/com/micabytes/util/FontHandler.java
package com.micabytes.util;

import android.content.Context;
import android.graphics.Typeface;
import android.util.SparseArray;

import com.micabytes.GameApplication;

@SuppressWarnings("UtilityClass")
public final class FontHandler {
  private static final SparseArray<Typeface> FONTS = new SparseArray<>();

  private FontHandler() {
    // NOOP
  }

  // Returns the hashcode of the font
  public static void loadFont(int key, String fontPath) {
    Context c = GameApplication.getInstance();
    Typeface type = Typeface.createFromAsset(c.getAssets(), fontPath);
    FONTS.append(key, type);
  }

  public static Typeface get(int key) {
    return FONTS.get(key);
  }

}

File: src/main/java/com/micabytes/util/FontHandler.kt
package com.micabytes.util

import android.graphics.Typeface
import com.micabytes.Game
import java.util.*


object FontHandler {
  private const val FONT_DIR = "fonts"
  private val CACHE = HashMap<String, Typeface>()

  operator fun get(fontName: String): Typeface? {
    return if (CACHE.containsKey(fontName)) {
      CACHE[fontName]
    } else {
      val typeface = Typeface.createFromAsset(Game.instance.assets, FONT_DIR + StringHandler.SLASH + fontName + ".ttf")
      CACHE[fontName] = typeface
      typeface
    }
  }

  /*
  @BindingAdapter("app:font")
  @JvmStatic
  fun setFont(view: TextView, fontName: String) {
    view.typeface = FontHandler[fontName]
  }

  @BindingAdapter("app:font")
  fun setFont(view: Button, fontName: String) {
    view.typeface = FontHandler[fontName]
  }
  */

}

